packages/data-point/src/DataPoint.js

```diff
@@ -0,0 +1,125 @@
+const { createReducer } = require("./create-reducer");
+const { Accumulator } = require("./Accumulator");
+const { resolve } = require("./resolve");
+const { Cache } = require("./Cache");
+const isPlainObject = require("./is-plain-object");
+
+/**
+ * Applies a reducer from an accumulator object.
+ *
+ * @param {Accumulator} acc accumulator object to be used.
+ * @param {Reducer} reducer
+ * @returns {Promise<any>} resolved value
+ */
+async function resolveFromAccumulator(acc, reducer) {
```

Comment:
this file has several async functions, but they don't use await - do they still need to be async?

Reply:
yes, async will wrap the functions on a promise, making any error inside them be thrown inside the promise chain, so for example in the case of 

```js
async function resolveFromAccumulator(acc, reducer) {
  const parsedReducers = createReducer(reducer);
  return resolve(acc, parsedReducers);
}
```

if createReducer produced an error, and the user was expecting to catch any error from the function via a catch the error would be thrown outside the promise would not have the expected behavior for catching the error, even though the function does return a promise.

... or am I wrong on this assumption, according to my tests it is the case